sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -127,13 +128,18 @@ trait InvokeLike extends Expression with NonSQLExpression {
       arguments: Seq[Expression],
       input: InternalRow,
       dataType: DataType): Any = {
-    val args = arguments.map(e => e.eval(input).asInstanceOf[Object])
-    if (needNullCheck && args.exists(_ == null)) {
+    var i = 0
+    val len = arguments.length
+    while (i < len) {
+      evaluatedArgs(i) = arguments(i).eval(input).asInstanceOf[Object]
```

Comment:
why we need to keep `evaluatedArgs`?

Reply:
You mean just use val?

Reply:
Doesn't we evaluate `arguments` for each time `invoke` is called? Why not just having `val evaluatedArgs: Array[Object] = new Array[Object](arguments.length)` in `invoke`?

Reply:
I guess it's aims to reuse `Array[Object]` itself and only changes the values of array.

Reply:
Yea even though we evaluate `arguments` for each `invoke` call we can reuse the same array to store the results of evaluation. I guess it's better than allocating a new `Array[Object]` for each input row.